library/snapshot.py

```diff
@@ -11,6 +15,169 @@
+            in the volume group. If neither snapshot_lvm_all_vgs or snapshot_lvm_set are set,
+            snapshot_lvm_vg is required. When used inside of a snapset definition, use
+            vg parameter.
+        type: str
+    snapshot_lvm_lv
+        description:
+            If set, the role will create snapshots for the single logical volume in the volume group
+            specified by snapshot_lvm_vg.  The parameter requires snapshot_lvm_vg is set to a valid
+            volume group. When used inside of a snapset definition, use lv parameter.
+        type: str
+    snapshot_lvm_verify_only
+        description:
+            If true, the check and remove commands verify that the system is in the correct state.
+            For the remove command, the target system will be searched for any snapshots that would
+            be removed by the remove command without snapshot_lvm_verify_only.
+        type: str
+    snapshot_lvm_mountpoint_create
+        description:
+            If the mount point specified doesn't currently exist, create the mount point and any
+            parent directories necessary for the mount point. When used inside of a snapset definition,
+            use mountpoint_create parameter.
+        type: bool
+    snapshot_lvm_mountpoint
+        description:
+            The mount target for the block device. When used inside of a snapset definition,
+            use mountpoint parameter.
+        type: str
+    snapshot_lvm_mount_origin
+        description:
+            If set to true, mount the origin of the snapshot rather than the snapshot.
+            When used inside of a snapset definition, use mount_origin parameter.
+        type: bool
+    snapshot_lvm_mount_options
+        description:
+            Options to pass to the mount command for the filesystem.  The argument is
+            a comma separated list.  See the man page for mount for details.
+            Note that XFS by default will not allow multiple filesystems with the
+            same UUID to be mounted at the same time.  Using the "nouuid" will
+            bypass the duplicate UUID check and allow a snapshot to be mounted
+            at the same time as the snapshot source.
+        type: str
+    snapshot_lvm_unmount_all
+        description:
+            If set to true, unmount all mountpoint for the resulting blockdevice.
+            Linux allows filesystems to be mounted in multiple locations.  Setting
+            this flag will unmount all locations.
+        type: bool
+    snapshot_lvm_vg_include
+        description:
+            When using `snapshot_lvm_all_vgs`, there may be some
+            subset of all volume groups that you want to use.  Set `snapshot_lvm_vg_include`
+            to a regex pattern that matches the names of the volume groups you want to use and
+            the rest will be excluded
+        type: str
+
+author:
+    - Todd Gill (tgill@redhat.com)
+"""
+
+
+EXAMPLES = r"""
+# Create Snapshots of all VGs
```

Comment:
This should be an example of using the module directly, not using the role.

Reply:
Ok.  The errors are a bit difficult to understand.   I was following the example in the storage role.  Currently I'm seeing a lot of errors for the docs: https://github.com/linux-system-roles/snapshot/actions/runs/9502379067/job/26190204498  - it looks like the storage role is using a different "-collection-version 1.78.2" vs the snapshot role is using "--collection-version 1.79.0"

Reply:
I also tried following the examples in: https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_general.html

But had trouble with errors with that format too.